docs/design.md

```diff
@@ -104,3 +104,17 @@ several ways according to policy:
 
 Then like the usual LibOS pattern it will call the container's entry point as though the container
 was a process unto itself (rather than where it now sits like a library in the LibOS PAL process).
+
+## Local Attestation for Filesystem Decryption Key Provisioning
```

Comment:
I think the best path for enclave-cc right now is to use sealing. There is some discussion in #3 subsequent to the creation of this PR to support the sealing direction.

Please update the heading to reflect that Local Attestation is exploratory or future work.

You will also want to update the design to discuss the policy. The text in the diagram of "check [agent|app] identity against policy" isn't sufficient to evaluate the design here. 

It is also important to define the APIs. 

I would also appreciate some discussion of the use of network interfaces, i.e. are these remotely accessible? How does the app-enclave address the agent-enclave?

Reply:
I think it would be fine to mark it as exploratory or future work. This doc just gives a design using TEE-agnostic approach to implement an attest channel to transfer a secret. Your questions are reasonable and could be addressed by the real implementation of this LA protocol. The discussion context is here: https://github.com/confidential-containers/enclave-cc/issues/3#issuecomment-1214284752

Reply:
In general, the additional detail I'm requesting is to distinguish the enclave-cc design from local attestation generally.